src/Alert.php

```diff
@@ -0,0 +1,644 @@
+    {
+        $new = clone $this;
+        $new->layoutHeader = $value;
+        return $new;
+    }
+
+    protected function run(): string
+    {
+        $new = clone $this;
+        return $new->renderAlert($new);
+    }
+
+    /**
+     * Render Alert.
+     */
+    private function renderAlert(self $new): string
+    {
+        $new->attributes['role'] = 'alert';
+        $new->attributes['id'] ??= $new->id ?? Html::generateId('alert-');
+
+        if (!isset($new->parts['{button}'])) {
+            $new->renderCloseButton($new);
+        }
+
+        if (!isset($new->parts['{icon}'])) {
+            $new->renderIcon($new);
+        }
+
+        if (!isset($new->parts['{body}'])) {
+            $new->renderBody($new);
+        }
+
+        if (!isset($new->parts['{header}'])) {
+            $new->renderHeader($new);
+        }
+
+        $contentAlert = $new->renderPanelHeader($new) . PHP_EOL . $new->renderPanelBody($new);
+
+        if ($new->class !== '') {
+            Html::addCssClass($new->attributes, $new->class);
+        }
+
+        return $new->body !== ''
+            ? Div::tag()
+                ->attributes($new->attributes)
+                ->content(PHP_EOL . trim($contentAlert) . PHP_EOL)
+                ->encode(false)
+                ->render()
+            : '';
+    }
+
+    /**
+     * Renders close button.
+     */
+    private function renderCloseButton(self $new): void
+    {
+        $new->parts['{button}'] = '';
+
+        if ($new->buttonOnClick !== '') {
+            $new->buttonAttributes['onclick'] = $new->buttonOnClick;
+        }
+
+        if ($new->buttonClass !== '') {
+            Html::addCssClass($new->buttonAttributes, $new->buttonClass);
+        }
+
+        $new->parts['{button}'] = PHP_EOL .
+            Button::tag()
+                ->attributes($new->buttonAttributes)
+                ->content($new->buttonLabel)
+                ->encode(false)
+                ->type('button')
+                ->render();
+    }
+
+    /**
+     * Render icon.
+     */
+    private function renderIcon(self $new): void
+    {
+        if ($new->iconClass !== '') {
+            Html::addCssClass($new->iconAttributes, $new->iconClass);
+        }
+
+        if ($new->iconContainerClass !== '') {
+            Html::addCssClass($new->iconContainerAttributes, $new->iconContainerClass);
+        }
+
+        $icon = CustomTag::name('i')->attributes($new->iconAttributes)->content($new->iconText)->render();
+
+        $new->parts['{icon}'] = PHP_EOL .
+            Div::tag()
+                ->attributes($new->iconContainerAttributes)
+                ->content($icon)
+                ->encode(false)
+                ->render() . PHP_EOL;
+    }
+
+    /**
+     * Render the alert message.
+     */
+    private function renderBody(self $new): void
+    {
+        if ($new->bodyClass !== '') {
+            Html::addCssClass($new->bodyAttributes, $new->bodyClass);
```

Comment:
Rendering should not modify attributes.

Reply:
https://github.com/terabytesoftw/yii-widgets/blob/7e1e5582d580b21b1c1b32afb8711612b950ea02/tests/AlertTest.php#L380 As you can see in the example, it is only configured if the user needs it, in this case we set the class of the body to render and that is correct, create the function so that it has ide support.

Reply:
Fixed.